process/drivers/traits.rs

```diff
@@ -12,17 +12,50 @@ use semver::{Version, VersionReq};
 
 use crate::drivers::{functions::get_private_key, types::CiDriverType, Driver};
 
+#[cfg(feature = "sigstore")]
+use super::sigstore_driver::SigstoreDriver;
 use super::{
+    buildah_driver::BuildahDriver,
+    cosign_driver::CosignDriver,
+    docker_driver::DockerDriver,
+    github_driver::GithubDriver,
+    gitlab_driver::GitlabDriver,
+    local_driver::LocalDriver,
     opts::{
         BuildOpts, BuildTagPushOpts, CheckKeyPairOpts, GenerateImageNameOpts, GenerateKeyPairOpts,
         GenerateTagsOpts, GetMetadataOpts, PushOpts, RunOpts, SignOpts, SignVerifyOpts, TagOpts,
         VerifyOpts, VerifyType,
     },
+    podman_driver::PodmanDriver,
+    skopeo_driver::SkopeoDriver,
     types::ImageMetadata,
 };
 
+trait PrivateDriver {}
+
+macro_rules! impl_private_driver {
+    ($driver:ty) => {
+        impl PrivateDriver for $driver {}
+    };
+}
+
+impl_private_driver!(Driver);
+impl_private_driver!(DockerDriver);
+impl_private_driver!(PodmanDriver);
+impl_private_driver!(BuildahDriver);
+impl_private_driver!(GithubDriver);
+impl_private_driver!(GitlabDriver);
+impl_private_driver!(LocalDriver);
+impl_private_driver!(CosignDriver);
+impl_private_driver!(SkopeoDriver);
+impl_private_driver!(CiDriverType);
```

Comment:
Repeat over the type so that we can have a comma delimited list of types instead